Description: Avoid implicit function definitions in the configure checks
 Add function prototypes and include system header files as needed.
Forwarded: not-yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2024-03-18

--- a/acinclude.m4
+++ b/acinclude.m4
@@ -994,6 +994,9 @@
 
 int resval = -1; /* -1: fail, 0: ok */
 
+void binsearch(long startval, long maxval);
+int checkval(long val);
+
 int
 main(void)
 {
@@ -1013,6 +1016,7 @@
    binsearch(0, MAXVAL);
 }
 
+void
 binsearch(long startval, long endval)
 {
    long midval = (endval - startval) / 2 + startval;
--- a/libscompat.m4
+++ b/libscompat.m4
@@ -34,6 +34,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <stdio.h>
+#include <string.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #ifndef SIOCGIFHWADDR
@@ -103,6 +104,7 @@
 
 AC_MSG_CHECKING([for sched_setaffinity])
 AC_TRY_COMPILE([
+#define _GNU_SOURCE
 #include <sched.h>
 ],
 [   cpu_set_t set1;
@@ -127,6 +129,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <sys/socket.h>
+#include <string.h>
 int main(void)
 {
     struct sockaddr_in addr;
@@ -303,6 +306,7 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <stdio.h>
 
 int
 main(void)
@@ -332,6 +336,7 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <stdio.h>
 
 int
 main(void)
@@ -359,6 +364,7 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <stdio.h>
 
 int
 main(void)
@@ -412,6 +418,7 @@
 AC_MSG_CHECKING([for CLOCK_MONOTONIC clock_gettime() support])
 AC_TRY_RUN([
 #include <sys/time.h>
+#include <stdio.h>
 #include <time.h>
 
 int
--- a/sharedopt.m4
+++ b/sharedopt.m4
@@ -101,8 +101,7 @@
 	#XXX include header in compilation test?
 	LIBS="$LIBS -lwrap"
 	AC_TRY_LINK([
-int allow_severity;
-int deny_severity;
+#include <tcpd.h>
 ], [hosts_access(0);],
         [AC_MSG_RESULT(yes)
 	 SOCKDDEPS="${SOCKDDEPS}${SOCKDDEPS:+ }-lwrap"
